### PR TITLE
support for extending the visibility timeout on per-message basis

### PR DESCRIPTION
**Pull request by demboos**
Proposal for a fix for [#92](https://github.com/spring-cloud/spring-cloud-aws/issues/92)

### CODE REVIEW

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/support/VisibilityHandlerMetodArgumentResolver.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.core.MethodParameter;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.handler.invocation.HandlerMethodArgumentResolver;
+import org.springframework.util.ClassUtils;
+
+/**
+ * @author Szymon Dembek
+ * @since 1.3
+ */
+public class VisibilityHandlerMetodArgumentResolver implements HandlerMethodArgumentResolver {
```
Comment (BernhardBln) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/support/VisibilityHandlerMetodArgumentResolver.java]: s/Metod/Method/ :)

Comment (demboos) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/support/VisibilityHandlerMetodArgumentResolver.java]: spelling mistake fixed, thanks for pointing out

**Comment by demboos** [comment]
I'm planning to add automatic (yet optional) visibility timeout extension too.

**Comment by alainsahli** [comment]
Hi @demboos 

Thank you very much for this pull request. We are going to integrate it when we start with milestone 1.3.

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageVisibility.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2013-2014 the original author or authors.
```
Comment (alainsahli) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageVisibility.java]: The date should be 2017

File: spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/listener/support/VisibilityHandlerMethodArgumentResolverTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package org.springframework.cloud.aws.messaging.listener.support;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+
+/**
+ * @author Alain Sahli
```
Comment (alainsahli) [inline spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/listener/support/VisibilityHandlerMethodArgumentResolverTest.java]: You should put your name ;-)

**Comment by yingqiaoott** [comment]
We would like to use this feature. When it would be merged to a release?

**Comment by aemruli** [comment]
@demboos thanks. Integrated into master (will be released with 2.0)

**Pull request merged**